File: modules/common/src/main/java/com/media/bus/common/exceptions/NoAuthorizationException.java
package com.media.bus.common.exceptions;

import com.media.bus.common.result.Result;
import com.media.bus.common.result.type.CommonResult;

import java.io.Serial;

/// 해당 기능 또는 자원에 권한이 없을 경우 throw
@SuppressWarnings("unused")
public class NoAuthorizationException extends BaseException {

	@Serial
    private static final long serialVersionUID = 2119428331469523918L;

	public NoAuthorizationException() {
		super(CommonResult.AUTHORIZATION_FAIL);
	}

	public NoAuthorizationException(String message) {
		super(CommonResult.AUTHORIZATION_FAIL, message);
	}

	public NoAuthorizationException(String message, Throwable cause) {
		super(CommonResult.AUTHORIZATION_FAIL, message, cause);
	}

	public NoAuthorizationException(Throwable cause) {
		super(CommonResult.AUTHORIZATION_FAIL, cause);
	}

	public NoAuthorizationException(Result result) {
		super(result);
	}

	public NoAuthorizationException(Result result, String message) {
		super(result, message);
	}

	public NoAuthorizationException(Result result, Throwable cause) {
		super(result, cause);
	}

	protected NoAuthorizationException(Result result, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
		super(result, cause, enableSuppression, writableStackTrace);
	}
}

File: modules/common/src/main/kotlin/com/media/bus/common/exceptions/NoAuthorizationException.kt
package com.media.bus.common.exceptions

import com.media.bus.common.result.Result
import com.media.bus.common.result.type.CommonResult

/** 해당 기능 또는 자원에 권한이 없을 경우 throw */
class NoAuthorizationException(
    result: Result = CommonResult.AUTHORIZATION_FAIL,
    message: String? = null,
    cause: Throwable? = null,
) : BaseException(result, message, cause) {

    companion object {
        private const val serialVersionUID: Long = 2119428331469523918L
    }
}
